### DISCUSSION TITLE
Changes in the Web API -  get details for the inverter

### DISCUSSION THREAD

**Discussion by Engehausen** [tbnobody/OpenDTU#1872] [Q&A]
Hello!

Previously you could use:
http://xxx.xxx.xxx.xxx/api/livedata/status/inverters
to read all the information about all the inverters. Now, to be able to read out the current power, the inverter number must also be entered after status. However, I would like to query several inverters so that the data is then available in an array as above (and earlier). Previously, the power could be read out directly during a query, e.g. using
        power0 = r['inverters'][0]['AC']['0']['Power']['v'] 
        power1 = r['inverters'][1]['AC']['0']['Power']['v'] 
If I specify an inverter number, only the detailed values of an inverter are output. How can I query both together in detail?
Would it be possible to join the request, like this: /api/livedata/status?inv=11418xxx&?inv=11417xxx

Thank you very much!
Peter

**Comment by ms49434** [comment]
Heap usage issues were the reason why the API was changed, see https://github.com/tbnobody/OpenDTU/commit/c27ecc3620f656fbdfaa7e9f0f3deca6a371a373
You can easily collect detailed information from all inverters by looping through the list.

**Reply by Engehausen** [reply]
What do you mean by looping through the list? How do I generate the list? I have now made a query for each inverter as a qickfix and then evaluate the data separately. But I don't see how I can use a query like http://xxx.xxx.xxx.xxx/api/livedata/status/inverters to create a list with all the data.

**Reply by MetaChuh** [reply]
@Engehausen

> I have now made a query for each inverter as a qickfix and then evaluate the data separately.

yes, this is the correct procedure 👍

looping just means to do this repeatedly, at the desired interval, which you already do.

greetings,
metachuh
